components/script/bluetooth_blacklist.rs

```diff
@@ -5,115 +5,117 @@
+pub fn uuid_is_blacklisted(uuid: &str, exclude_type: Blacklist) -> bool {
+    BLUETOOTH_BLACKLIST.with(|blist| {
+        match exclude_type {
+            Blacklist::All => {
+                blist.borrow().is_blacklisted(uuid)
+            },
+            Blacklist::Reads => {
+                blist.borrow().is_blacklisted_for_reads(uuid)
+            }
+            Blacklist::Writes => {
+                blist.borrow().is_blacklisted_for_writes(uuid)
+            }
         }
-    }
+    })
 }
 
-pub struct BluetoothBlacklist(Option<HashMap<String, ExcludeType>>);
+pub struct BluetoothBlacklist(Option<HashMap<String, Blacklist>>);
 
 #[derive(Eq, PartialEq)]
-pub enum ExcludeType {
-    Exclude, // Read and Write
-    ExcludeReads,
-    ExcludeWrites,
+pub enum Blacklist {
+    All, // Read and Write
+    Reads,
+    Writes,
 }
 
 impl BluetoothBlacklist {
-
     // https://webbluetoothcg.github.io/web-bluetooth/#blacklisted
     pub fn is_blacklisted(&self, uuid: &str) -> bool {
         match self.0 {
-            Some(ref map) => map.get(uuid).map_or(false, |et| et.eq(&ExcludeType::Exclude)),
+            Some(ref map) => map.get(uuid).map_or(false, |et| et.eq(&Blacklist::All)),
             None => false,
         }
     }
 
     // https://webbluetoothcg.github.io/web-bluetooth/#blacklisted-for-reads
     pub fn is_blacklisted_for_reads(&self, uuid: &str) -> bool {
         match self.0 {
-            Some(ref map) => map.get(uuid).map_or(false, |et| et.eq(&ExcludeType::Exclude) ||
-                                                              et.eq(&ExcludeType::ExcludeReads)),
+            Some(ref map) => map.get(uuid).map_or(false, |et| et.eq(&Blacklist::All) ||
+                                                              et.eq(&Blacklist::Reads)),
             None => false,
         }
     }
 
     // https://webbluetoothcg.github.io/web-bluetooth/#blacklisted-for-writes
     pub fn is_blacklisted_for_writes(&self, uuid: &str) -> bool {
         match self.0 {
-            Some(ref map) => map.get(uuid).map_or(false, |et| et.eq(&ExcludeType::Exclude) ||
-                                                              et.eq(&ExcludeType::ExcludeWrites)),
+            Some(ref map) => map.get(uuid).map_or(false, |et| et.eq(&Blacklist::All) ||
+                                                              et.eq(&Blacklist::Writes)),
             None => false,
         }
     }
 }
 
 // https://webbluetoothcg.github.io/web-bluetooth/#parsing-the-blacklist
-fn parse_blacklist() -> Option<HashMap<String, ExcludeType>> {
+fn parse_blacklist() -> Option<HashMap<String, Blacklist>> {
     // Step 1 missing, currently we parse ./resources/gatt_blacklist.txt.
-    let file = match File::open(FILE_PATH) {
-        Ok(file) => file,
-        Err(_) => return None,
-    };
-    let content = BufReader::new(file);
     let valid_uuid_regex = Regex::new(VALID_UUID_REGEX).unwrap();
+    let mut data: Vec<u8> = Vec::new();
```

Comment:
The reason I put this data variable outside from the method is the following error: https://gist.github.com/zakorgy/41163c357fe896b1d2db197d0dccc022
In sort the lines method is only implemented to `&[u8]`, which type needs a lifetime specifier. (We can't implement it for a `Vec<u8>`, because it's a built in function in a built in trait. )But if I create the `data` variable in the `read_resource_file_lines`, which must hold the data for the returned `Lines` iterator to iterate, it does not live long enough after the function call finish.